src/main/java/edu/kit/ipd/crowdcontrol/objectservice/crowdworking/PlatformManager.java

```diff
@@ -150,13 +152,22 @@ public Payment getPlatformPayment(String name) {
                 .map(platform1 -> platform1.publishTask(experiment))
                 .orElseThrow(() -> new IllegalArgumentException("Platform not found!"))
                 .handle((s1, throwable) -> {
-                    if (s1 != null) {
+                    //if the creation was successful update the task
+                    if (s1 != null && throwable == null) {
                         result.setPlatformData(s1);
-                    } else {
-                        result.setStatus(TaskStatus.stopped);
+                        if (!tasksOps.updateTask(result)) {
+                            throw new IllegalStateException("Updating record for published task failed");
+                        }
+                    }
+                    //if there is no useful key throw!
+                    if (s1 == null || s1.isEmpty()) {
```

Comment:
I just figured out that if an exception occured, s1 is indeed always null, no matter if it was assigned a value or not. So you were right! That means the third "if block" will never be executed. So it might be a good idea to combine the last two "if blocks". I think it is important to pass or handle the errors that occur on the platform connection level.